src/dotnet/commands/dotnet-new3/HelpFormatter.cs

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.DotNet.Tools.New3
+{
+    public class HelpFormatter
+    {
+        public static HelpFormatter<T> For<T>(IEnumerable<T> items, int columnPadding, char? headerSeparator = null, bool blankLineBetweenRows = false)
+        {
+            return new HelpFormatter<T>(items, columnPadding, headerSeparator, blankLineBetweenRows);
+        }
+    }
+
+    public class HelpFormatter<T>
+    {
+        private readonly bool _blankLineBetweenRows;
+        private readonly int _columnPadding;
+        private readonly List<ColumnDefinition> _columns = new List<ColumnDefinition>();
+        private readonly char? _headerSeparator;
+        private readonly IEnumerable<T> _items;
+        private readonly List<Tuple<int, bool, IComparer<string>>> _ordering = new List<Tuple<int, bool, IComparer<string>>>();
+
+        public HelpFormatter(IEnumerable<T> items, int columnPadding, char? headerSeparator, bool blankLineBetweenRows)
```

Comment:
`items` could use a null check so that we don't get a `NullReferenceException` when `Layout` is called.